### PR TITLE
feat(HACBS-586): report status when GitHub webhook integration is used

### PR DESCRIPTION
**Pull request by amisstea**
Refactored much of the git and status packages to improve testability.

Signed-off-by: Alex Misstear <amisstea@redhat.com>

### CODE REVIEW

File: git/github/github.go [hunk 1]
```diff
@@ -48,95 +48,146 @@ type ChecksService interface {
...
+
+// GetAppsService returns either the default or custom Apps service.
+func (c *Client) GetAppsService() AppsService {
+	if c.apps == nil {
+		return c.gh.Apps
+	}
+	return c.apps
+}
+
+// GetAppsService returns either the default or custom Checks service.
```
Comment (dirgim) [inline git/github/github.go:90]: nit: GetAppsService ->  GetChecksService

File: git/github/github.go [hunk 2]
```diff
@@ -48,95 +48,146 @@ type ChecksService interface {
...
 
-// AppClient is an abstraction around the github API client.
-type AppClient struct {
-	logger         logr.Logger
-	client         *ghapi.Client
-	appID          int64
-	installationID int64
-	apps           AppsService
-	checks         ChecksService
+// GetAppsService returns either the default or custom Repositories service.
```
Comment (dirgim) [inline git/github/github.go:106]: nit: GetAppsService -> GetRepositoriesService

File: status/status.go [hunk 3]
```diff
@@ -61,11 +61,7 @@ func (a *Adapter) GetReporters(ctx context.Context, pipelineRun *tektonv1beta1.P
 	var reporters []Reporter
 
 	if helpers.HasLabelWithValue(pipelineRun, gitops.PipelineAsCodeGitProviderLabel, gitops.PipelineAsCodeGitHubProviderType) {
-		reporter, err := a.githubReporter(ctx, a.logger, a.k8sClient, pipelineRun)
```
Comment (dirgim) [inline status/status.go:64]: Looks like `ctx` parameter won't be needed anymore for the `GetReporters` function.

File: git/github/github.go (hunk 1 above)
Comment (amisstea) [reply git/github/github.go]: Fixed.

File: git/github/github.go [hunk 4]
```diff
@@ -48,95 +48,146 @@ type ChecksService interface {
...
+	}
+	return c.checks
+}
 
-// AppClientInterface defines the methods that should be implemented by a GitHub App client
-type AppClientInterface interface {
-	CreateCheckRun(context.Context, *CheckRunAdapter) (*int64, error)
-	UpdateCheckRun(context.Context, int64, *CheckRunAdapter) error
-	GetCheckRunID(context.Context, string, string, string, string) (*int64, error)
+// GetAppsService returns either the default or custom Issues service.
```
[the remark this answers is not in the document]
Comment (amisstea) [reply git/github/github.go]: Fixed.

File: git/github/github.go (hunk 2 above)
Comment (amisstea) [reply git/github/github.go]: Fixed.

File: status/status.go (hunk 3 above)
Comment (amisstea) [reply status/status.go:64]: Removed.

Comment (dirgim) [reply status/status.go:64]: Thanks!

**Review by jsztuka — APPROVED** [review]
lgtm

File: status/reporters.go [hunk 5]
```diff
@@ -172,37 +226,185 @@ func (r *GitHubPipelineRunReporter) createCheckRunAdapter() (*github.CheckRunAda
...
-	if !helpers.HasLabelWithValue(r.pipelineRun, gitops.PipelineAsCodeEventTypeLabel, gitops.PipelineAsCodePullRequestType) {
+// When using GitHub webhook integration a commit status and, in some cases, a comment is created.
+func (r *GitHubReporter) ReportStatus(ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) error {
+	if !helpers.HasLabelWithValue(pipelineRun, gitops.PipelineAsCodeEventTypeLabel, gitops.PipelineAsCodePullRequestType) {
 		return nil
 	}
 
-	if r.installationID > 0 {
-		checkRun, err := r.createCheckRunAdapter()
+	if helpers.HasAnnotation(pipelineRun, gitops.PipelineAsCodeInstallationIDAnnotation) {
```
Comment (dirgim) [inline status/reporters.go:362]: minor suggestion: adding a comment explaining that the `gitops.PipelineAsCodeInstallationIDAnnotation` is used to identify that the GitHub App integration is in place for the PaC pipelineRun.

**Review by dirgim — APPROVED** [review]
LGTM aside the minor suggestion

File: status/reporters.go (hunk 5 above)
Comment (amisstea) [reply status/reporters.go:364]: Good point. Comment added.

Comment (dirgim) [reply status/reporters.go:364]: Thanks!

**Pull request merged**